### PR TITLE
issue 265 Map Events via RootScope need to be unique for Multiple Maps

### PR DESCRIPTION
**Pull request by nmccready**
refactor(leafletEvents): leafletEvents is gone as a service; you must now use the explicity Events service helpers.

refactor(rootScope map events): The map id is no included in the
leafletWhatecer event if the leaflet directives id is set.

Example: `<leaflet id="1" >`
Will fire  leafletDirectiveMap.1.click ..
leafletDirectiveMap.1.eventName .

leafletDirectiveMarker.1.click ...

For most people with a single map this will remain the same
leafletDirectiveMarker.click or leafletDirectiveMap.click.

- spec for testing map event handling when leaflet id is defined

- examples to double check evens mods to markers, geojson, and paths

issue #265

### CODE REVIEW

File: test/unit/eventsDirectiveSpec.js [hunk 1]
```diff
@@ -62,22 +130,26 @@ describe('Directive: leaflet', function() {
...
+                map.fireEvent([origEventName]);
+                expect(check[origEventName]).not.toEqual(true);
+            });
         });
-    });
 
-    it('should NOT broadcast map events from the rootscope if the event-broadcast option is not an object',function() {
+      });
+
+    xit('should NOT broadcast map events from the rootscope if the event-broadcast option is not an object',function() {
```
Comment (nmccready) [inline test/unit/eventsDirectiveSpec.js]: `xit` a ton of places to show specs that are doing nothing the proper way

File: test/unit/eventsDirectiveSpec.js [hunk 2]
```diff
@@ -20,7 +21,74 @@ describe('Directive: leaflet', function() {
...
+          }
+
+          if(mapId)
+            mapId = mapId + '.';
+
+          leafletData.getMap().then(function(map) {
+              mapEvents.forEach(function(origEventName){
+                  var eventName = 'leafletDirectiveMap.' + mapId + origEventName;
+                  // leafletLogger.log(eventName); // Inspect
+                  scope.$on(eventName, function(){
```
Comment (nmccready) [inline test/unit/eventsDirectiveSpec.js]: Notice setEventTrue is wrapped in a function now as the original spec was generating false positives.

File: test/unit/bootstrap.coffee [hunk 3]
```diff
@@ -2,12 +2,16 @@ window.ngLeafLetTestGlobals = {}
...
-    angular.module('leaflet-directive').config ($provide) ->
+    angular.module('leaflet-directive')
+    .config ($provide) ->
+        $provide.value('$log', console)
         $provide.decorator '$timeout', ($delegate, $browser) ->
             $delegate.hasPendingTasks = ->
                 $browser.deferredFns.length > 0
 
             $delegate
+    .run (leafletLogger) ->
```
Comment (nmccready) [inline test/unit/bootstrap.coffee]: Get more logging going to help w debugging specs.

File: test/unit/bootstrap.coffee [hunk 4]
```diff
@@ -2,12 +2,16 @@ window.ngLeafLetTestGlobals = {}
 
 beforeEach ->
 
-    angular.module('leaflet-directive').config ($provide) ->
+    angular.module('leaflet-directive')
+    .config ($provide) ->
+        $provide.value('$log', console)
```
Comment (nmccready) [inline test/unit/bootstrap.coffee]: Enable specs to log via  provider

File: src/services/events/leafletMapEvents.js [hunk 5]
```diff
@@ -81,10 +80,19 @@ angular.module("leaflet-directive")
         }
     };
 
+    var _addEvents =  function(map, mapEvents, contextName, scope, logic){
```
Comment (nmccready) [inline src/services/events/leafletMapEvents.js]: moved from leafletEvent which has been fully removed and was deprecated

**Comment by nmccready** [comment]
@zacronos @elesdoar @jessertaylor @tombatossals @simison please review

**Comment by zacronos** [comment]
LGTM

**Comment by elesdoar** [comment]
:+1:

**Pull request merged**